xknxproject/models/knxproject.py

```diff
@@ -92,6 +92,29 @@ class Space(TypedDict):
     project_uid: int | None
     devices: list[str]
     spaces: dict[str, Space]
+    functions: list[Function]
+
+
+class Function(TypedDict):
+    """Function typed dict."""
+
+    identifier: str
+    name: str
+    function_type: str
+    usage_text: str
+    project_uid: int | None
+    group_addresses: list[GroupAddressRef]
+
+
+class GroupAddressRef(TypedDict):
+    """GroupAddressRef typed dict."""
+
+    identifier: str
+    name: str
+    role: str
+    ref_id: str
```

Comment:
Do you actually need this ref_id? We are resolving the GA anyway..

Reply:
I removed the ref_id.

tox runs locally and in the CI. The problem occurs in the "Create coverage report" step. But I have no idea what I'm doing wrong here.

Reply:
No, it just looks like it does. CI just passes without doing anything 🤷
Coverage report fails since it can't find a coverage file then

Reply:
Ok, that was a big mistake on my part. I accidentally deleted the tox.ini. As a result, it ran through without errors, but also did nothing.